bootable-jar/WFLY-14411_Bootable_Jar_Upgrade.adoc

```diff
@@ -0,0 +1,246 @@
+
+* An artifact present in the  `overridden-server-artifacts` or `overridden-artifacts` set must be present in the dependencies section. 
+It is advised that the dependency scope be `provided` to be not included in the application artifact (eg: war).  A warning is displayed if
+the scope is not `provided`.
+If an artifact is not present in the dependencies, then a failure occurs during build.
+
+* An artifact is referenced in the `overridden-server-artifacts` by GroupId, ArtifactId and optionally Classifier. 
+Version being retrieved from the Maven dependencies. 
+
+* An artifact is referenced in the `overridden-artifacts` by GroupId, ArtifactId, optionally Classifier and optionally Version. 
+Version is being retrieved from the Maven dependencies if not set. 
+
+* An overridden artifact (artifact present in the `overridden-server-artifacts` list) must be unique. Any duplicate will make the packaging to fail.
+
+* An overridden artifact (artifact present in the `<FeaturePack>` `overridden-artifacts` list) must be unique. Any duplicate will make the packaging to fail.
+
+* Adding an overridden artifact that is not part of the provisioned server artifacts will lead to a failure during build. If the overridden artifact is associated
+to a feature-pack, the check is done in the context of the feature-pack.
+
+* Adding an overridden Galleon feature-pack artifact that is not a dependency of the WildFly server being 
+provisioned will lead to an error during packaging. 
+
+* The version of the galleon feature-packs referenced using GAV becomes optional in the maven plugin configuration if the  Galleon 
+feature-pack artifact has been added to the dependencies. In this case the version is being retrieved from the dependencies. If the version
+is missing and no Feature-pack artifact is retrieved in the dependencies, a failure occurs during packaging.
+
+* Third party galleon feature-packs benefit from this upgrade capability for JBoss Module artifacts they are bringing to the provisioned server.
+
+* The server jar components that can be upgraded are:
+** The JBoss module runtime jar (jboss-modules.jar file).
+** All jar artifacts referenced from JBoss Modules modules.
+
+* The set of artifacts that can be upgraded can be retrieved by setting the parameter `<dump-original-artifacts>true</dump-original-artifacts> or 
+the system property `bootable.jar.dump.original.artifacts` to true when building a bootable JAR. 
+The file `target/bootable-jar-build-artifacts/bootable-jar-server-original-artifacts.xml` is generated. 
+It contains XML elements for the Galleon feature-packs dependencies, JBoss Modules runtime and artifacts. 
+JBoss Modules modules artifacts are grouped by JBoss Modules name.
+The generated file contains only the artifacts that are provisioned by Galleon. 
+Each artifact version is the one that would get installed when building the Bootable JAR without upgrade.
```

Comment:
If artifact transformation is done when building the bootable jar (a la what is done to produce EE 9 compatible artifacts for WildFly Preview), the generated file contains the identifier of the untransformed artifact, not the transformed artifact which may have a different Version.

If ^^^ is accurate I suggest adding it.

Reply:
Thank-you. Added.